lib/tasks/orchestration.rake

```diff
@@ -0,0 +1,76 @@
+# TRANSLATORS: do not translate
+desc <<-END_DESC
+  Orchestration maintainance tasks.
+
+  WARNING: Always backup server data (e.g. DHCP leases file, DNS journal files)
+  prior running these tasks. They do nothing by default, unless perform=1 is
+  specified.
+
+  Examples:
+  rake orchestration:dhcp:add_missing subnet_name=NAME
+    Preview missing DHCP records on a DHCP Smart Proxy.
+
+  rake orchestration:dhcp:add_missing subnet_name=NAME perform=1
+    Do create missing DHCP records on a DHCP Smart Proxy.
+
+  rake orchestration:dhcp:remove_offending subnet_name=NAME
+    Preview offending DHCP records on a DHCP Smart Proxy.
+
+  rake orchestration:dhcp:remove_offending subnet_name=NAME perform=1
+    Do remove offending DHCP records on a DHCP Smart Proxy.
+END_DESC
+
+def fetch_subnet(subnet_name)
+  subnet = Subnet.unscoped.find_by_name(subnet_name)
+  raise("Subnet '#{subnet_name}' not found") unless subnet
+  raise("Subnet '#{subnet_name}' has no DHCP proxy associated") unless subnet.dhcp_proxy
+  subnet
+end
+
+def fetch_dhcp_records(subnet)
+  subnet.dhcp_proxy.subnet(subnet.network)["reservations"].map do |rec|
+    Net::DHCP::Record.new(rec.merge(proxy: subnet.dhcp_proxy, network: subnet.network))
+  end
+end
+
+namespace :orchestration do
+  namespace :dhcp do
+    task :add_missing => :environment do
+      User.as_anonymous_admin do
+        dry_run = !ENV['perform']
```

Comment:
should this be `ENV['perform'] != '1'` just to make sure? because `!'0' == false` as well

Reply:
Good catch, ouch. Rebased.